backend/MoleculeOSSite/Models/Validators/RegisterValidator.cs

```diff
@@ -0,0 +1,21 @@
+﻿using FluentValidation;
+using MoleculeOSSite.Entities;
+
+namespace MoleculeOSSite.Models.Validators
+{
+    public class RegisterValidator:AbstractValidator<User>
+    {
+        public RegisterValidator(MyDbContext dbcontext)
+        {
+            RuleFor(x=>x.Email).NotEmpty().EmailAddress();
+            RuleFor(x => x.PasswordHash).NotEmpty().MinimumLength(6);
+            RuleFor(x=>x.ConfirmPassword).Equal(u=>u.PasswordHash);
+            RuleFor(x => x.Email).Custom((value, context) =>
+            {
+                var emailExist = dbcontext.Users.Any(u=>u.Email == value);
+                if (emailExist)
+                    context.AddFailure("Email", "Email is already taken");
```

Comment:
Tutaj pojawił się pewien problem - jeśli dwie osoby wpadną na to, aby użyć tego samego username to im się uda to. Co bądź będą mieli inne emaile, ale jeśli ktoś z nich postanowi się zalogować za pomocą username to może dojść do sytuacji, gdzie dostanie on token tej drugiej osoby zamiast swój. Jeśli zalogowałby się on przez email - tego problemu nie ma (ponieważ jest tutaj wymuszone użycie unikatowego emailu przy rejestracji), ale jeśli ktoś zaloguje się przez username a w bazie jest kilku takich użytkowników o takim samym username to nie ma gwarancji tego, że otrzyma on swój token, jest prawdopodobieństwo, że dostanie czyjś token.